defender/downloaded/Contract_cToken.js

```diff
@@ -243,80 +241,156 @@ async function postToDiscord(discordWebhook, message) {
     content: message,
   };
 
-  // perform the POST request
-  const response = await axios({
+  const discordObject = {
     url: discordWebhook,
     method: 'post',
     headers,
-    data: JSON.stringify(body),
-  });
-
+    data: body,
+  };
+  let response;
+  try {
+    // perform the POST request
+    response = await axios(discordObject);
+  } catch (err) {
+    if (err.response && err.response.status === 429) {
+      // rate-limited, retry
+      // after waiting a random amount of time between 2 and 120 seconds
+      const delay = getRandomInt(2000, 120000);
+      // eslint-disable-next-line no-promise-executor-return
+      const promise = new Promise((resolve) => setTimeout(resolve, delay));
+      await promise;
+      response = await axios(discordObject);
+    } else {
+      throw err;
+    }
+  }
   return response;
 }
 
+function getAddressForMatchReason(reason, logs, abi) {
+  let found;
+  let parsedLog;
+  const { params, signature } = reason;
+  const iface = new ethers.utils.Interface(abi);
+  const keys = Object.keys(params);
+  let key;
+  let value;
+  for (let i = 0; i < logs.length; i++) {
+    try {
+      // attempt to parse the log
+      // because there could be many logs emitted by many contracts
+      // we don't know whether the call to parseLog will fail, hence
+      // wrapping it with a try...catch
+      // if parseLog throws, we will move on to the next log
+      parsedLog = iface.parseLog(logs[i]);
+
+      // check that the event signature matches
+      // this should be equivalent to checking the topic hashes
+      if (parsedLog.signature !== signature) {
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+
+      // check that all of the parameter names and values match
+      // the keys and values in the params variable are the named
+      // arguments and their associated values for the event
+      // the parsedLog Object will have its named arguments and values
+      // stored under the args key
+      found = true;
+      for (let j = 0; j < keys.length; j++) {
+        key = keys[j];
+        value = params[key];
+        if (parsedLog.args[key].toString() !== value.toString()) {
```

Comment:
This will not work when the parameter name collides with a reserved word in JS as ethers.js will not add reserved words to `args`. However, I don't believe any of the cToken ABIs contain reserved words, so in this case it _should_ work.

Reply:
If Defender is also using ethers.js under the hood, then perhaps the filtering will be equally applied on both the `parsedLog.args` Object and the `params` Object?  (semi-speculative, but perhaps true)